### PR TITLE
From/ToSql for chrono types.

### PR DESCRIPTION
**Pull request by gwenn**

### CODE REVIEW

**Comment by euclio** [comment]
@gwenn @jgallagher Any progress on merging this? Serializing to string and back is a huge pain point for me.

File: src/types/chrono.rs [hunk 1]
```diff
@@ -0,0 +1,374 @@
...
+/// ("YYYY-MM-DDTHH:MM:SS.SSS[+-]HH:MM"/"YYYY-MM-DDTHH:MM"/"YYYY-MM-DDTHH:MM:SS"/"YYYY-MM-DDTHH:MM:SS.SSS" also supported)
+/// When the timezone is not specified, Local is used.
+impl FromSql for DateTime<Local> {
+    unsafe fn column_result(stmt: *mut sqlite3_stmt, col: c_int) -> Result<DateTime<Local>> {
+        match sqlite3_column_type(stmt, col) {
+            ffi::SQLITE_TEXT => {
+                let s = try!(String::column_result(stmt, col));
+                if s.len() > 23 {
+                    let fmt = if s.as_bytes()[10] == b'T' {
+                        "%Y-%m-%dT%H:%M:%S%.f%:z"
```
Comment (iwinux) [inline src/types/chrono.rs]: Is `:z` a typo (should be `%:z`)? Trying to parse `"2016-05-04 17:00:42.862471+08:00"` with `"%Y-%m-%d %H:%M:%S%.f%:z"`, got error `input contains invalid characters`.

Comment (gwenn) [inline src/types/chrono.rs]: Indeed !
```rust
        // Worst
        UTC.datetime_from_str("2016-05-04 17:00:42.862471+08:00", "%Y-%m-%d %H:%M:%S%.f%:z").unwrap_err();
        UTC.datetime_from_str("2016-05-04 17:00:42.862471Z", "%Y-%m-%d %H:%M:%S%.f%:z").unwrap_err();
        UTC.datetime_from_str("2016-05-04 17:00:42.862471", "%Y-%m-%d %H:%M:%S%.f%:z").unwrap_err();

        UTC.datetime_from_str("2016-05-04 17:00:42.862471+08:00", "%Y-%m-%d %H:%M:%S%.f%z").unwrap_err();
        UTC.datetime_from_str("2016-05-04 17:00:42.862471Z", "%Y-%m-%d %H:%M:%S%.f%z").unwrap_err();
        UTC.datetime_from_str("2016-05-04 17:00:42.862471", "%Y-%m-%d %H:%M:%S%.f%z").unwrap_err();

        // Better but...
        DateTime::parse_from_str("2016-05-04 17:00:42.862471+08:00", "%Y-%m-%d %H:%M:%S%.f%z").unwrap().with_timezone(&UTC);
        DateTime::parse_from_str("2016-05-04 17:00:42.862471Z", "%Y-%m-%d %H:%M:%S%.f%z").unwrap_err(); // Invalid
        DateTime::parse_from_str("2016-05-04 17:00:42.862471", "%Y-%m-%d %H:%M:%S%.f%z").unwrap_err(); // TooShort

        DateTime::parse_from_str("2016-05-04 17:00:42.862471+08:00", "%Y-%m-%d %H:%M:%S%.f%:z").unwrap().with_timezone(&UTC);
        DateTime::parse_from_str("2016-05-04 17:00:42.862471Z", "%Y-%m-%d %H:%M:%S%.f%:z").unwrap_err(); // Invalid
        DateTime::parse_from_str("2016-05-04 17:00:42.862471", "%Y-%m-%d %H:%M:%S%.f%:z").unwrap_err(); // TooShort

        // Best but in SQLite, the timezone indicator is optional
        DateTime::parse_from_rfc3339("2016-05-04T17:00:42.862471+08:00").unwrap().with_timezone(&UTC);
        DateTime::parse_from_rfc3339("2016-05-04T17:00:42.862471Z").unwrap().with_timezone(&UTC);
        DateTime::parse_from_rfc3339("2016-05-04T17:00:42.862471").unwrap_err(); // TooShort

        "2016-05-04T17:00:42.862471+08:00".parse::<DateTime<UTC>>().unwrap();
        "2016-05-04T17:00:42.862471Z".parse::<DateTime<UTC>>().unwrap();
        "2016-05-04T17:00:42.862471".parse::<DateTime<UTC>>().unwrap_err(); // TooShort
```

**Comment by jgallagher** [comment]
Very sorry about the lack of activity. I have time this week to get this (and the other open) PRs reviewed and merged in for a new release.

@gwenn I need to review this one a little more carefully still, but one thing that jumps out at me is the assumption that floats should be converted to/from Julian dates. I know SQLite supports that, but I think that's relatively unusual (e.g., it mostly uses `strftime`'s substitutions but added `%J` for Julian date). I would've expected floats to convert to/from fractional seconds since 1970. Even if it's slightly less natural inside SQLite itself, it's still supported:

```
sqlite> SELECT (julianday('now') - 2440587.5)*86400.0;
1463341804.65901
sqlite> SELECT strftime('%Y-%m-%d %H:%M:%f', 1463341804.65901, 'unixepoch');
2016-05-15 19:50:04.659
```

Another alternative is to not support converting ints/floats at all, and instead only provide implementations that serialize to and from strings (that seems the least error prone since SQLite handles them fine and they're immediately obvious to a person inspecting the database). What do you think?

**Pull request merged**